Trappist/src/Promact.Trappist.Repository/Profile/ProfileRepository.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Promact.Trappist.DomainModel.DbContext;
+using Promact.Trappist.Web.Models;
+using Microsoft.AspNetCore.Identity;
+using System.Threading.Tasks;
+using Promact.Trappist.DomainModel.Models;
+using System;
+
+namespace Promact.Trappist.Repository.Profile
+{
+    public class ProfileRepository : IProfileRepository
+    {
+        private readonly TrappistDbContext _dbContext;
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public ProfileRepository(TrappistDbContext dbContext, UserManager<ApplicationUser> userManager)
+        {
+            _dbContext = dbContext;
+            _userManager = userManager;
+        }
+
+        /// <summary>
+        /// fetch the profile details of the cureent user 
+        /// </summary>
+        /// <param name="name">takes parameter of string type which has the name of the current user whose details have to be fetched</param>
+        /// <returns>details of the current user</returns>
+        public async Task<ApplicationUser> GetUserDetailsAsync(string name)
+        {
+            var user = await _userManager.FindByNameAsync(name);
+            return user;
+        }
+
+        /// <summary>
+        /// Updates current user details
+        /// </summary>
+        /// <param name="updateUserDetails">parameter of type ApplicationUser which has the updated details of the user profile</param>
+        /// <returns>Update user profile details in database</returns>
+        public void UpdateUserProfile(ApplicationUser updateUserDetails)
+        {
+            var user = _dbContext.Users.Update(updateUserDetails);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "user" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)